src/python/pants/backend/terraform/tool.py

```diff
@@ -388,7 +415,30 @@ async def setup_terraform_process(
     )
     env = await Get(EnvironmentVars, EnvironmentVarsRequest(terraform.extra_env_vars))
 
-    immutable_input_digests = {"__terraform": downloaded_terraform.digest}
+    extra_bins = await Get(
+        BinaryShims,
+        BinaryShimsRequest,
+        BinaryShimsRequest.for_paths(getent_binary, rationale="download terraform providers"),
+    )
+
+    path = []
+    user_path = env.get("PATH")
+    if user_path is not None:
```

Comment:
Edge case, but you probably don't want to append a blank `PATH` value either. Maybe just use `if user_path:` here then?

Reply:
good point! We should centralise merging envvars at some point, I think there are a few subsystems that allow a passthrough of envvars in some way

Reply:
I started doing something like this in the adhoc / shell backends via the [`prepare_env_vars` function](https://github.com/pantsbuild/pants/blob/fcd3642f2ec833f63b74f981204d194a70b8f891/src/python/pants/core/util_rules/adhoc_process_support.py#L626-L682)

Reply:
Parts of that method might be useful elsewhere in the future.